### ISSUE TITLE
Cannot add any device

### ISSUE THREAD

**Issue by DanielRitter75** [rospogrigio/localtuya#1470]
## The problem
In dialog Configure Tuya device I always get "An unknown error occurred. See log for details".
See log below.

## Environment
Raspberry Pi 4 
Web-Access via Chrome on Windows 10
Android-Access via Home Assistant App

- Localtuya version: latest 
- Home Assistant Core version: 
Home Assistant 2023.7.3
Supervisor 2023.07.1
Operating System 10.3
Frontend 20230705.1 - lates
- [x] Does the device work using the Home Assistant Tuya Cloud component ? 
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [] Was the device working with earlier versions of localtuya ? Which one? 
- [] Are you using the Tuya/SmartLife App in parallel ? 

## Provide Home Assistant traceback/logs

```
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.localtuya.config_flow
Source: custom_components/localtuya/pytuya/__init__.py:1182
Integration: LocalTuya integration (documentation, issues)
First occurred: 21:10:49 (1 occurrences)
Last logged: 21:10:49

Unexpected exception: [Errno 113] Connect call failed ('192.168.3.135', 6668)
Traceback (most recent call last):
  File "/config/custom_components/localtuya/config_flow.py", line 595, in async_step_configure_device
    self.dps_strings = await validate_input(self.hass, user_input)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/config_flow.py", line 242, in validate_input
    interface = await pytuya.connect(
                ^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 1182, in connect
    _, protocol = await loop.create_connection(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1085, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1069, in create_connection
    sock = await self._connect_sock(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 973, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 634, in sock_connect
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/selector_events.py", line 674, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
OSError: [Errno 113] Connect call failed ('192.168.3.135', 6668)

```

**Comment by dekela** [comment]
I am having the same issue with the exact specs

**Comment by null-forbidden** [comment]
I am also having this issue after one of my lights got factory reset. Now I am unable to add that one.

**Comment by Biomassfreak** [comment]
I'm having the same problem, really hoping resetting the router isn't the solution.

**Comment by royaldunlin** [comment]
Just cycling the power to the smart bulb was enough to resolve this error for me.

**Comment by inijman** [comment]
Same here. Trying to move from cloud tuya in HA (where all devices worked nicely) to local tuya. No devices where discovered, and manually adding devices resulted in the error above. My Tuya cloud account expires tomorrow but I saved all information but if this is not working I loose a lot of devices.

**Comment by svyatoslav00012** [comment]
same issue

**Comment by Singleton-95** [comment]
Had a same error today when I wanted to setup localtuya. 

Tried few debugging methods, from tcpdump I got ARP request from HA but not response from device. When I manually add ARP table entry everything works.

Not sure why are Tuya device not responding with ARP replay, and I am not sure if router factory reset could help... Regarding resetting device, it may help (?) but also it could stop responding at any time.

I have a really dirty solution which I am not a fun of, but it solves my problem for now. In my case all devices periodically broadcast from port 6667 (Tuya discovery port) which I used to fill ARP table with following Python script:


        from scapy.all import *

        PACKET_FILTER = "dst host 255.255.255.255 and dst port 6667"
        ARP_INSERT_COMMAND = "arp -i eth0 -s %s %s"

        ALREADY_PROCESSED = dict()

        def packet_handler(packet):
                try:
                        ipPacket = packet["IP"]
                        sourceIpAddress = ipPacket.src

                        etherPacket = packet["Ether"]
                        sourceMacAddress = etherPacket.src

                        if sourceIpAddress not in ALREADY_PROCESSED or ALREADY_PROCESSED[sourceIpAddress] != sourceMacAddress:
                                print('Adding pair (%s, %s) to ARP table' % (sourceIpAddress, sourceMacAddress))

                                command = ARP_INSERT_COMMAND  % (sourceIpAddress, sourceMacAddress)
                                os.system(command);

                                ALREADY_PROCESSED[sourceIpAddress] = sourceMacAddress
                except Exception as ex:
                        print(ex);


        sniff(filter = PACKET_FILTER, prn = packet_handler)

Hope it could help some of you.

**Comment by Geoffrey54** [comment]
> Had a same error today when I wanted to setup localtuya.
> 
> Tried few debugging methods, from tcpdump I got ARP request from HA but not response from device. When I manually add ARP table entry everything works.
> 
> Not sure why are Tuya device not responding with ARP replay, and I am not sure if router factory reset could help... Regarding resetting device, it may help (?) but also it could stop responding at any time.
> 
> I have a really dirty solution which I am not a fun of, but it solves my problem for now. In my case all devices periodically broadcast from port 6667 (Tuya discovery port) which I used to fill ARP table with following Python script:
> 
> ```
>     from scapy.all import *
> 
>     PACKET_FILTER = "dst host 255.255.255.255 and dst port 6667"
>     ARP_INSERT_COMMAND = "arp -i eth0 -s %s %s"
> 
>     ALREADY_PROCESSED = dict()
> 
>     def packet_handler(packet):
>             try:
>                     ipPacket = packet["IP"]
>                     sourceIpAddress = ipPacket.src
> 
>                     etherPacket = packet["Ether"]
>                     sourceMacAddress = etherPacket.src
> 
>                     if sourceIpAddress not in ALREADY_PROCESSED or ALREADY_PROCESSED[sourceIpAddress] != sourceMacAddress:
>                             print('Adding pair (%s, %s) to ARP table' % (sourceIpAddress, sourceMacAddress))
> 
>                             command = ARP_INSERT_COMMAND  % (sourceIpAddress, sourceMacAddress)
>                             os.system(command);
> 
>                             ALREADY_PROCESSED[sourceIpAddress] = sourceMacAddress
>             except Exception as ex:
>                     print(ex);
> 
> 
>     sniff(filter = PACKET_FILTER, prn = packet_handler)
> ```
> 
> Hope it could help some of you.

I have the same error here.

You say that the port is 6667 when in the log it says that the application try to connect to port 6668 :
`Unexpected` exception: [Errno 110] Connect call failed ('192.168.1.13', 6668)

**Comment by Singleton-95** [comment]
@Geoffrey54

HA integration is trying to connect to port 6668, but Tuya devices are broadcasting on 6667 ([here](https://ask.wireshark.org/question/31629/does-anyone-know-what-comunication-can-this-be/) is a little bit of details).

NOTE: It could be possible that port depends on firmware I also found online discussions about port 6666.

Here is tcpdump

![tcpdump](https://github.com/rospogrigio/localtuya/assets/105790199/9c872dcc-2c67-4699-b722-5536e1521d0f)

**Comment by Geoffrey54** [comment]
I'm still stuck with that issue, is there anybody with that issue that can help ?

**Comment by barrelltitor** [comment]
I have the same problem, I am unable to add any new devices

Unexpected exception: [Errno 113] Connect call failed ('192.168.88.149', 6668)

**Comment by Geoffrey54** [comment]
> I have the same problem, I am unable to add any new devices
> 
> Unexpected exception: [Errno 113] Connect call failed ('192.168.88.149', 6668)

Have you the official tuya integration up with all the device running as me ?

**Comment by barrelltitor** [comment]
@Geoffrey54 Yeah, however it started working for me. 

What I did:

Restart Router
Restart the system HA is running on

This did not fix anything, however it might have contributed. After powering the device on/off and trying to add it as soon as it powered on and connected to wifi, it worked. 

Could be that the other integration is blocking things somehow, though I haven't tried disabling it

**Comment by lowerclasswarfare** [comment]
I had this problem earlier to which I commented in another issue. #1513 

I thought something might be amiss when I couldn't use the Jellyfin integration. I pinged my Jellyfin server from Home Assistant within the network and wasn't getting any response. Anyways, It seemed to be a firewall issue on my router. I passed all ipv4 traffic between local interfaces and now I can talk to those devices.
